source/run-command/run-command.rst

```diff
@@ -187,8 +193,152 @@ Drivers MUST document the behavior of RunCommand if a ``maxTimeMS`` field  is al
 * See Client Side Operations Timeout's section on `runCommand <https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst#runcommand>`_
 * See Client Side Operations Timeout's section on `runCommand behavior <https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst#runcommand-behavior>`_
 
+
+--------------------
+``runCursorCommand``
+--------------------
+
+Drivers MAY expose a runCursorCommand API with the following syntax.
+
+.. code:: typescript
+
+    interface Database {
+      /**
+       * Takes an argument representing an arbitrary BSON document and executes it against the server.
+       */
+      runCursorCommand(command: BSONDocument, options: RunCursorCommandOptions): RunCommandCursor;
+    }
+
+    interface RunCursorCommandOptions extends RunCommandOptions {
+      /**
+       * This option is an enum with possible values CURSOR_LIFETIME and ITERATION.
+       * For operations that create cursors, timeoutMS can either cap the lifetime of the cursor or be applied separately to the original operation and all subsequent calls.
+       * To support both of these use cases, these operations MUST support a timeoutMode option.
+       *
+       * @defaultValue CURSOR_LIFETIME
+       *
+       * @see https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst
+       */
+      timeoutMode?: ITERATION | CURSOR_LIFETIME;
+
+      /**
+       * See the `cursorType` enum defined in the crud specification.
+       * @see https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#read
+       *
+       * Identifies the type of cursor this is.
+       *
+       * A tailable cursor can receive empty `nextBatch` arrays in `getMore` responses.
+       * However, subsequent `getMore` operations may return documents if new data has become available.
+       *
+       * A tailableAwait cursor is an enhancement where instead of dealing with empty responses the server will block until data becomes available.
+       *
+       * @defaultValue NON_TAILABLE
+       */
+      cursorType: CursorType;
```

Comment:
The current pattern in the Go Driver (i.e. GODRIVER-1605) is to allow the user to set batchSize (and now maxTimeMS and comment) _after_ the runCommand has been executed. This pattern means these settings only affect subsequent document batches fetched from the database. Will that pattern work for cursorType as well?

Reply:
cursorType is here to supply CSOT implementations with the information they need to correctly support timeoutMode. We do not have a driver that supports timeoutMode yet so cursorType is also not necessary, it communicates information about the initial command being run rather than anything that would change how to run/construct getMores.

I improved the docs here mentioning the relation to CSOT, and that this field is an optional option. 

cc: @dariakp

Reply:
Will we be creating follow-up driver tickets to implement cursorType and timeoutMode? Per your comment, the Go team would like to implement them simultaneously if possible.